File: C++/add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        while (num >= 10) {
            num = sumDigits(num);
        }
        return num;
    }
    
    int sumDigits(int x) {
        int sum = 0;
        while (x) {
            sum += x % 10;
            x /= 10;
        }
        return sum;
    }
};

File: Kotlin/add-digits.kt
class Solution {
    fun addDigits(num: Int): Int {
        var tmp = num
        while (tmp >= 10) {
            var sum = 0
            while (tmp > 0) {
                sum += tmp % 10
                tmp /= 10
            }
            tmp = sum
        }
        return tmp
    }
}